torch/csrc/jit/script/compiler.cpp

```diff
@@ -1086,38 +1218,57 @@ struct to_ir {
 
     return expr_value;
   }
-
-  Value* emitCond(const Expr& cond) {
-    Value* v = emitExpr(cond);
+  Value* emitToBool(Value* v) {
+    SourceRange loc = v->node()->sourceRange();
     Value* out;
     try {
-      auto bool_cast = environment_stack->getSugaredVar("bool", cond.range());
-      out = asSimple(bool_cast->call(cond.get()->range(), method, {v}, {}, 0));
+      auto bool_cast = environment_stack->getSugaredVar("bool", loc);
+      out = asSimple(bool_cast->call(loc, method, {v}, {}, 0));
     } catch (...) {
-      throw ErrorReport(cond.range()) << "Could not cast value of type "
-                                      << v->type()->python_str() << " to bool";
+      throw ErrorReport(loc) << "Could not cast value of type "
+                             << v->type()->python_str() << " to bool";
     }
     // cast value not response for checking output type
     if (!out->type()->isSubtypeOf(BoolType::get())) {
-      throw ErrorReport(cond)
+      throw ErrorReport(loc)
           << "expected a bool expression for condition but found "
           << out->type()->python_str();
     }
     return out;
   }
 
-  void emitIfElseBlocks(Value* cond_value, const If& stmt) {
-    Node* n = graph->insertNode(create(prim::If, stmt.range(), 0));
-    n->addInput(cond_value);
-    const auto bool_info = findRefinements(stmt.cond());
+  void emitIfElseBlocks(
+      const SourceRange& loc,
+      const CondValue& cond_value,
+      const List<Stmt>& trueBranch,
+      const List<Stmt>& falseBranch) {
+    // this is a static if statement: that is, it contains a subset
+    // of operators where we are willing to specialize the if statement
+    // to be only the true or false branch when the condition is statically
+    // known. This is used to meta-program modules, for instance, when a
+    // submodule is absent, an is None check can be used to ensure the
+    // accesses to the None check, which would error, are not compiled.
+    if (cond_value.staticIf()) {
+      if (*cond_value.staticIf()) {
```

Comment:
Not compiling one branch of an if statement with multiple `staticIf` checks is a new feature so maybe add tests for that